applications/GeoMechanicsApplication/custom_utilities/stress_strain_utilities.cpp

```diff
@@ -152,4 +152,26 @@ Matrix StressStrainUtilities::CalculateGreenLagrangeStrainTensor(const Matrix& r
                   IdentityMatrix(rDeformationGradient.size1()));
 }
 
+Vector StressStrainUtilities::CalculateCauchyStrain(const Matrix& rB, const Vector& rDisplacements)
+{
+    return prod(rB, rDisplacements);
+}
+
+std::vector<Vector> StressStrainUtilities::CalculateStrains(const std::vector<Matrix>& rDeformationGradients,
+                                                            const std::vector<Matrix>& rBs,
+                                                            const Vector& rDisplacements,
+                                                            bool          UseHenckyStrain,
+                                                            std::size_t   VoigtSize)
+{
+    std::vector<Vector> result;
+    std::transform(
+        rDeformationGradients.begin(), rDeformationGradients.end(), rBs.begin(), std::back_inserter(result),
+        [&rDisplacements, UseHenckyStrain, VoigtSize](const auto& rDeformationGradient, const auto& rB) {
+        return UseHenckyStrain ? CalculateHenckyStrain(rDeformationGradient, VoigtSize)
+                               : CalculateCauchyStrain(rB, rDisplacements);
+    });
+
+    return result;
+}
```

Comment:
I'm still not too sure if the stress strain utilities should be responsible of choosing which strain measure to calculate based on a flag (also since we only provide two possibilities here, while there are other possibilities in this file e.g. VonMisesStrain). I was probably a bit ambiguous during our discussion. Maybe it's good to double check this with @WPK4FEM and @avdg81?

I do agree that we put the choice directly in the transform instead of as a separate function though.

Reply:
I discussed a related question with @WPK4FEM to use both CalculateStrains and CalculateAnyOfMaterialResponse at the same time. He has suggested to keep it now.

Reply:
To not delay this PR, let's keep it the way it is for now. We may want to revisit this functionality later when we'd like to add support for more strain measures.